pkg/apis/core/validation/shoot.go

```diff
@@ -932,6 +932,7 @@ func validateNetworking(networking *core.Networking, workerless bool, fldPath *f
 	}
 
 	primaryIPFamily := helper.DeterminePrimaryIPFamily(networking.IPFamilies)
+	apiServerProxyRange := cidrvalidation.NewCIDR(v1beta1constants.ReservedRangeApiServerProxy, field.NewPath(""))
```

Comment:
Please check how the overlap with the VPN ranges is done and do the same for this instead of checking here.

The question is how to deal with existing clusters. Those should not break completely.